### PR TITLE
Add multi-platform builds

### PR DESCRIPTION
**Pull request by natescherer**
Hello!
This PR adds a platform input to both the GitHub Action and AzDO Task to allow mutli-platform builds. I've tested each successfully. I've also done my best to minimally modify the structure of the action/task.

Fixes part of #151.

Please let me know if you have any questions/changes!

### CODE REVIEW

**Comment by natescherer** [comment]
Not sure what's going on here? I've run build-local.sh multiple times and it doesn't show any changes, but your CI buil test shows uncommitted changes to azdo-task/DevcontainersCi/dist/index.js.

**Comment by stuartleeks** [comment]
That's odd! I just grabbed the code and ran `build-local.sh` on my machine and pushed a commit with the change.

**Comment by stuartleeks** [comment]
Thanks @natescherer - this looks really interesting.

Based on the discussion in https://github.com/devcontainers/ci/issues/150, it looks as though the `platform` input can be used alongside `runCmd`, is that correct? 

It would be great to have a test that includes `platform` and `runCmd` to ensure that combination is validated going forwards.

**Comment by natescherer** [comment]
Yep, this was designed to have runCmd still work when using platform (with caveats mentioned in the multi-platform-builds.md file I added).

Tests for both GH and AzDo added.

File: .github/workflows/ci_common.yml [hunk 1]
```diff
@@ -756,3 +756,52 @@ jobs:
         uses: ./
         with:
           subFolder: github-tests/Dockerfile/build-only
+
+  test-platform-with-runcmd:
```
Comment (stuartleeks) [inline .github/workflows/ci_common.yml:760]: Can you also add this job to the `needs` list in the `pr-checks` job?

Comment (natescherer) [reply .github/workflows/ci_common.yml:760]: Done!

**Comment by stuartleeks** [comment]
I just merged `main` in and re-ran `build-local.sh` to resolve the conflicts. Hope you don't mind!

**Comment by natescherer** [comment]
> I just merged `main` in and re-ran `build-local.sh` to resolve the conflicts. Hope you don't mind!

Not at all! We're all working towards the same goal here.

File: github-action/src/main.ts [hunk 2]
```diff
@@ -46,6 +48,17 @@ export async function runMain(): Promise<void> {
 		);
 		const userDataFolder: string = core.getInput('userDataFolder');
 
+		if (platform) {
+			const skopeoInstalled = await isSkopeoInstalled();
+			if (!skopeoInstalled) {
+				core.warning(
```
Comment (chrmarti) [inline github-action/src/main.ts:54]: Is there a way to avoid the dependency on an additional CLI?

**Review by chrmarti — COMMENTED** [review]
Thanks for the PR! Didn't do a detailed review, but left a comment.

File: github-action/src/main.ts (hunk 2 above)
Comment (natescherer) [reply github-action/src/main.ts:54]: So, yes and no. The docker cli can do a build and push in one step, but then the ability to run the intermediate runCmd is lost. (There's no way to just push with the docker cli that I was able to find). I figured there was value in still being able to use runCmd on multi-platform builds, which is why I went with the strategy of building to an OCI tar, running the runCmd, and then, if it succeeds, doing the push via skopeo.

FWIW, skopeo is included by default in all runner images.

**Comment by natescherer** [comment]
Looks like that latest merge from main meant build-local.sh needed ran again, which was blocking tests.

**Comment by natescherer** [comment]
Oops, it looks like I forgot to revert my AzDO task version bump from when I was testing out the extension, sorry about that. This should get past that build bug.

File: github-action/src/main.ts (hunk 2 above)
Comment (natescherer) [reply github-action/src/main.ts:54]: Hi @chrmarti, sorry for the ping but just wanted to make sure you saw my response. Thanks!

Comment (chrmarti) [reply github-action/src/main.ts:54]: Thanks for the ping and the detailed response! If it's installed by default, we're fine for now. Thanks!

**Review by chrmarti — APPROVED** [review]
LGTM!

**Comment by Chuxel** [comment]
Thanks for this! Really appreciate the multi-repo PR and thought you put into it!

**Pull request merged**